v1/lists.yaml

```diff
@@ -0,0 +1,1011 @@
+        description: "Last modification date (in ISO 8601)"
+      order:
+        type: array
+        description: All IDs of the entries of the list in the user-defined order
+        items:
+          type: integer
+        example: [1, 3, 2]
+        uniqueItems: true
+        maxItems: *maxEntriesPerList
+  list_entry_write: &list_entry_write
+    type: object
+    properties:
+      <<: *list_entry_common
+    required:
+      - project
+      - title
+paths:
+  /lists/setup:
+    post:
+      tags:
+        - Reading lists
+      summary: Opt in to use reading lists.
+      description: |
+        Must precede other list operations.
+
+        Request must be authenticated with a MediaWiki session cookie.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      produces:
+        - application/json; charset=utf-8
+      parameters:
+        - name: csrf_token
+          in: query
+          required: true
+          schema: *csrf_token
+      responses:
+        '200':
+          description: Success.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
+              body:
+                action: readinglists
+                command: setup
+                token: '{{request.query.csrf_token}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8
+                cache-control: "no-cache"
+      x-monitor: false
+  /lists/teardown:
+    post:
+      tags:
+        - Reading lists
+      summary: Opt out from using reading lists.
+      description: |
+        Deletes all data. User needs to opt in again before being able to do anything.
+
+        Request must be authenticated with a MediaWiki session cookie.
+
+        Stability: [unstable](https://www.mediawiki.org/wiki/API_versioning#Unstable)
+      produces:
+        - application/json; charset=utf-8
+      parameters:
+        - name: csrf_token
+          in: query
+          required: true
+          schema: *csrf_token
+      responses:
+        '200':
+          description: Success.
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/problem'
+      x-route-filters:
+        - path: ./lib/mediawiki_auth_filter.js
+      x-request-handler:
+        - forward_to_mw:
+            request:
+              uri: /{domain}/sys/action/rawquery
+              body:
+                action: readinglists
+                command: teardown
+                token: '{{request.query.csrf_token}}'
+            return:
+              status: 200
+              headers:
+                content-type: application/json; charset=utf-8
+                cache-control: "no-cache"
```

Comment:
Did you mean(TM): `cache-control: max-age=0, s-maxage=0` ?

Reply:
AIUI `cache-control: max-age=0, s-maxage=0` means "should actually send the request", `cache-control: no-cache` means "must actually send the request". This is a POST endpoint that changes data and returns a success flag, so the latter seems more appropriate.

Reply:
According to http://www.faqs.org/rfcs/rfc2616.html section 9.5 POST responses are only cached if appropriate `cache-control` headers are present, which means that if the header is absent the response shouldn't be cached meaning that this is not needed.